src/visualization/types/Map/view.tsx

```diff
@@ -61,53 +76,60 @@ const GeoPlot: FC<Props> = ({result, properties}) => {
     return ''
   }
 
-  switch (errorCode) {
-    case MapErrorStates.MapServiceUnavailable:
-      error =
-        'Map type is having issues connecting to the server. Please try again later.'
-
-      return (
-        <div className="panel-resizer--error" data-testid="geoplot-error">
-          {error}
-        </div>
-      )
-    case MapErrorStates.InvalidCoordinates:
-      error =
-        'Map type is not supported with the data provided: Missing latitude/longitude values'
-
-      return (
-        <div className="panel-resizer--error" data-testid="geoplot-error">
-          {error}
-        </div>
-      )
-    default:
-      const tileServerConfiguration = {
-        tileServerUrl: getMapboxUrl(),
-        bingKey: '',
-      }
+  if (
+    mapServiceError === RemoteDataState.NotStarted ||
+    coordinateError === RemoteDataState.NotStarted ||
+    mapServiceError === RemoteDataState.Loading ||
+    coordinateError === RemoteDataState.Loading
+  ) {
+    return null
+  }
 
-      return (
-        <Plot
-          config={{
-            table: result.table,
-            showAxes: false,
-            layers: [
-              {
-                type: 'geo',
-                lat: calculatedGeoCoordinates.lat,
-                lon: calculatedGeoCoordinates.lon,
-                zoom,
-                allowPanAndZoom,
-                detectCoordinateFields: true,
-                mapStyle,
-                layers,
-                tileServerConfiguration: tileServerConfiguration,
-              },
-            ],
-          }}
-        />
-      )
+  if (mapServiceError === RemoteDataState.Error) {
+    error =
+      'Map type is having issues connecting to the server. Please try again later.'
+
+    return (
+      <div className="panel-resizer--error" data-testid="geoplot-error">
+        {error}
+      </div>
+    )
+  } else if (coordinateError === RemoteDataState.Error) {
+    error =
+      'Map type is not supported with the data provided: Missing latitude/longitude values (field values must be specified to lat or lon)'
+
+    return (
+      <div className="panel-resizer--error" data-testid="geoplot-error">
+        {error}
+      </div>
+    )
+  }
+  const tileServerConfiguration = {
+    tileServerUrl: getMapboxUrl(),
+    bingKey: '',
```

Comment:
is this a holdover from bonitoo's work?

Reply:
So if bingKey is provided, it renders bing maps. If not, it renders the map url we send it.

Reply:
fancy!